Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.py

```diff
@@ -0,0 +1,304 @@
+"""
+
+"""
+
+# python 3.9 imports
+from functools import wraps
+from json import dumps, loads
+
+# accessdata imports
+from accessdata.api.agents import Agent
+from accessdata.api.jobs import Job
+from accessdata.api.filters import and_, or_
+from accessdata.client import Client
+
+# xsoar imports
+from CommonServerPython import (
+    CommandResults,
+    return_values,
+    return_results
+)
+import demistomock as demisto
+
+""" decorator wrapping demisto commands"""
+
+_run_functions = {}
+
+def wrap_demisto_command(command):
+    def _func(func):
+        @wraps(func)
+        def _inside(*args, **kwargs):
+            return func(*args, **kwargs)
+        _run_functions[command] = func
+        return _inside
+    return _func
+
+""" register demisto commands """
+
+@wrap_demisto_command("accessdata-api-get-case-by-name")
+def _get_case_by_name(client, name):
+    case = client.cases.first_matching_attribute("name", name)
+    if not case:
+        demisto.error(f"Failed to gather case with name ({name}).")
+
+    return CommandResults(
+        outputs_prefix="Accessdata.Case",
+        outputs={
+            "ID": case["id"],
+            "Name": name,
+            "CaseFolder": case["ftkcasefolderpath"]
+        }
+    )
+
+@wrap_demisto_command("accessdata-api-create-case")
+def _create_case(client, **kwargs):
+    case = client.cases.create(**kwargs)
+    if not case:
+        demisto.error(f"Failed to create case.")
+        return CommandResults(
+            outputs_prefix="Accessdata.Case",
+            outputs={}
+        )
```

Comment:
Change the way you take care of errors. Raise errors in command functions, and catch it at the main function. 
```suggestion
    if not case:
        raise ValueError(f"Failed to create case.")

```

Reply:
Code has been updated to make use of Exceptions throughout with try except in the main function body.